Material_FAB/src/main/ets/components/MainPage/ripple/ripple.ets

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 Application Library Engineering Group.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RippleModel } from './rippleModel'
+
+@Component
+export struct Ripple {
+  @State model: RippleModel.Model = undefined
+  @Link private show_ripple : boolean
+
+  private show_animation() {
+    animateTo({
+      duration: this.model.getRippleDuration(),
```

Comment:
better to add validation on model object, recheck for required validations in other files as well.